### DISCUSSION TITLE
FTDI Can't flash firmware or new bootloader - Tried it all, any suggestions?

### DISCUSSION THREAD

**Discussion by gfalgiano** [pascallanger/DIY-Multiprotocol-TX-Module#612] [Q&A]
I have a Vantac module that I use with an X-Lite Pro. It suffers from the issue of not starting up correctly unless I power cycle the radio or switch models back and forth. I'm trying to update the bootloader to fix the issue.

I took apart the module and soldered some wires to the Pins on the board and connected to an FTDI adapter. I also soldered a bridge between the Boot and 3.3v.

I can't get multi flash or Arduino to read or write to the board. I was able to do this a few weeks back with the same equipment, but no luck this time. I do see a solid green LED when I bridge the Boot and 3.3v pin. It seems like that was the main fix for someone else who had a similar issue.

What else I tried:

- Reversed TX and RX wires
- Confirmed COM port (only one shows up at a time)
- Checked connectivity multiple times
- Changed USB cable
- Restarted computer
- Powered from USB instead of 3.3v on FTDI adapter
- Tried with and without boot to 3.3v bridge and even doing it at different times when powered…
- Re-soldered all connections and checked connectivity out to the pins on the FTDI adapter.
- Tried different drivers for the FTDI adapter
- Tried different Baud rates
- Tried flashing with Arduino, same settings that worked for me previously, but now says “Failed to init device”
Sketch uses 46828 bytes (84%) of program storage space. Maximum is 55272 bytes.
Global variables use 3776 bytes (18%) of dynamic memory, leaving 16704 bytes for local variables. Maximum is 20480 bytes.

Looking for Maple device ...
No MULTI-Module device found.
Probing serial port COM5 for STM32 in BOOT0 mode ...
stm32flash 0.4
Failed to init device.
http://stm32flash.googlecode.com/
Interface serial_w32: 115200 8E1
Module in BOOT0 mode not found on COM5 

- Tried flashing firmware via FTDI with multi flash v6, v5 and v3 via FTDI, but says “Failed to read the multi-module”
Verbose output:
Reading MULTI-Module via serial
[1/1] Reading flash memory ... failed!
.\tools\stm32flash.exe -b 115200 -r "C:\Users\admin\AppData\Local\Temp\tmpF6EA.tmp" COM5
Failed to init device.
stm32flash 0.4
http://stm32flash.googlecode.com/
Interface serial_w32: 115200 8E1

- Tried flashing bootloader with USB connection instead of the FTDI adapter and it reads the module and writes the firmware (says successful), but the red light doesn't go out after the initial flashing of the LED. Let it sit for 10min, no change. Seems like it isn’t applying the bootloader.

I’m able to flash via USB and SD card from the radio, just can’t change the bootloader or flash via FTDI adapter. With those methods, it loads new firmware fine, but then I still have the boot issue I described at the beginning.

Any help is greatly appreciated!

**Comment by pascallanger** [comment]
What happens when you shunt Boot0 to 3.3V? Does the module starts normally meaning you can control a plane?

**Reply by gfalgiano** [reply]
@pascallanger 
It turns the LED solid green and the red LED off. This is the same pattern I see normally when I fly. I didn't try controlling a plane though. 

However, when I plug in USB with the shunt Boot0 to 3.3V I get a windows error.

**Comment by pascallanger** [comment]
Can you try controlling a plane or simply see if in the protocol menu you still have the firmware version appearing when your Boot0 to 3.3V shunt is in place?

**Reply by gfalgiano** [reply]
@pascallanger I tried controlling a model with the Boot0 and 3.3v shorted and I cannot control the model. On the radio, the status shows as "No Telemetry".

I power cycled and changed models a couples times to verify, but it won't connect. Connects fine without the short though.

**Reply by pascallanger** [reply]
Ok so it means that you are entering the bootloader mode correctly.
So basically there is not 30 things:
- verify that GND is well connected between the serial adapter and the module
- verify that TX of the module is well connected to the RX serial adapter
- verify that RX of the module is well connected to the TX serial adapter
- try to upload
- if that fails then I would say that you have damaged the STM32 somehow... May be your serial adapter was set to 5V at one point?

**Reply by gfalgiano** [reply]
I checked and my FTDI adapter is only putting out 3.15v

I tried another adapter and still no success.

I guess it is time to order an iRangex
